jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/CachedContentFactory.java

```diff
@@ -293,10 +294,10 @@ private void shrinkCache()
             // Scan the entire cache and generate an ordered list by last accessed time.
             SortedSet<CachedHttpContent> sorted = new TreeSet<>((c1, c2) ->
             {
-                if (c1._lastAccessed < c2._lastAccessed)
+                if (c1._lastAccessed.isBefore(c2._lastAccessed))
                     return -1;
 
-                if (c1._lastAccessed > c2._lastAccessed)
+                if (c1._lastAccessed.isAfter(c2._lastAccessed))
                     return 1;
```

Comment:
```java
int cmp = c1._lastAccessed.compareTo(c2._lastAccessed);
if (cmp != 0)
    return cmp;
cmp = c1._contentLengthValue - c2._contentLengthValue;
if (cmp != 0)
    return cmp < 0 ? -1 : 1;
return c1._key.compareTo(c2._key);
```

Note how the implementation was broken because `_contentLengthValue` was only tested for `<` -- in contrast `_lastAccessed` was tested for both `<` and `>`.

Reply:
So, `_contentLengthValue` is a long, the resulting code would be ...

``` java
SortedSet<CachedHttpContent> sorted = new TreeSet<>((c1, c2) ->
{
    int cmp = c1._lastAccessed.compareTo(c2._lastAccessed);
    if (cmp != 0)
        return cmp;

    cmp = Long.compare(c1._contentLengthValue, c2._contentLengthValue);
    if (cmp != 0)
        return cmp;

    return c1._key.compareTo(c2._key);
});
```

Which could be simplified to ...

``` java
SortedSet<CachedHttpContent> sorted = new TreeSet<>(
    Comparator.comparing((CachedHttpContent c) -> c._lastAccessed)
        .thenComparingLong(c -> c._contentLengthValue)
        .thenComparing(c -> c._key));
```

Reply:
Done, using comparator chain instead.

Reply:
Nice!